src/GafferScene/Shader.cpp

```diff
@@ -213,6 +213,18 @@ class Shader::NetworkBuilder
 			return m_network;
 		}
 
+		const ValuePlug *parameterSource( const IECoreScene::ShaderNetwork::Parameter &parameter )
+		{
+			for( auto &[shader, handleAndHash] : m_shaders )
+			{
+				if( handleAndHash.handle == parameter.shader )
+				{
+					return shader->parametersPlug()->getChild<ValuePlug>( parameter.name );
```

Comment:
I think we want `descendant()` here rather than `getChild()`, so we can get parameters inside structs.